avm/ptn/deployment-script/import-image-to-acr/tests/e2e/max/dependencies.bicep

```diff
@@ -0,0 +1,160 @@
+@description('Optional. The location to deploy to.')
+param location string = resourceGroup().location
+
+@description('Required. The name of the Virtual Network to create.')
+param virtualNetworkName string
+
+@description('Required. The name of the Managed Identity to create.')
+param managedIdentityName string
+
+@description('Required. The name of the Azure Container Registry to create.')
+param acrName string
+
+@description('Required. The name of the Storage Account to create.')
+param storageAccountName string
+
+var registryRbacRoles = ['7f951dda-4ed3-4680-a7ca-43fe172d538d', '8311e382-0749-4cb8-b61a-304f252e45ec'] // ArcPull, AcrPush
+var storageAccountRbacRole = '69566ab7-960f-475b-8e7c-b3118f30c6bd' // Storage File Data Privileged Contributor
+
+module identity 'br/public:avm/res/managed-identity/user-assigned-identity:0.2.1' = {
+  name: managedIdentityName
+  params: {
+    name: managedIdentityName
+    location: location
+    enableTelemetry: false
+  }
+}
+
+// networking related resources
+resource vnet 'Microsoft.Network/virtualNetworks@2023-11-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: ['10.0.0.0/16']
+    }
+  }
+  resource subnet_privateendpoints 'subnets@2023-11-01' = {
+    name: 'privateendpoints-subnet'
+    properties: {
+      addressPrefix: '10.0.0.0/24'
+    }
+  }
+  resource subnet_deploymentscript 'subnets@2023-11-01' = {
+    name: 'deploymentscript-subnet'
+    dependsOn: [subnet_privateendpoints]
+    properties: {
+      addressPrefix: '10.0.1.0/24'
+      serviceEndpoints: [
+        {
+          service: 'Microsoft.Storage'
+        }
+      ]
+      delegations: [
+        {
+          name: 'Microsoft.ContainerInstance.containerGroups'
+          properties: {
+            serviceName: 'Microsoft.ContainerInstance/containerGroups'
+          }
+        }
+      ]
+    }
+  }
+}
+
+module dnsZoneContainerRegistry 'br/public:avm/res/network/private-dns-zone:0.2.5' = {
+  name: '${uniqueString(deployment().name, location)}-dnsZone-ACR'
+  params: {
+    name: 'privatelink.azurecr.io'
+    virtualNetworkLinks: [
+      {
+        name: '${vnet.name}-ContainerRegistry-link'
+        virtualNetworkResourceId: vnet.id
+        registrationEnabled: false
+      }
+    ]
+  }
+}
+
+module privateEndpointContainerRegistry 'br/public:avm/res/network/private-endpoint:0.4.1' = {
```

Comment:
Why not deploying the private endpoint as part of the registry module?

Reply:
They should be there & work:
https://github.com/Azure/bicep-registry-modules/blob/c0a08742d1e8e89577158452c191b39340814dba/avm/res/container-registry/registry/main.bicep#L404-L455

Reply:
It just does most of the heavy lifting for you as per this example:
https://github.com/Azure/bicep-registry-modules/blob/c0a08742d1e8e89577158452c191b39340814dba/avm/res/container-registry/registry/tests/e2e/waf-aligned/main.test.bicep#L98-L105

Reply:
That did work. Thanks for the hint.